### ISSUE TITLE
New v4.0 does not support Java 8

### ISSUE THREAD

**Issue by cordahi** [bsorrentino/maven-annotation-plugin#82]
The new v4.0 [unconditionally](https://github.com/bsorrentino/maven-annotation-plugin/blob/v4.0/processor/src/main/java/org/bsc/maven/plugin/processor/AbstractAnnotationProcessorMojo.java#L571) specifies the `--module-path` compiler option which is only supported in Java 9 in contradiction of the first line of the [README.md](https://github.com/bsorrentino/maven-annotation-plugin/blob/v4.0/README.md#a-maven-plugin-to-process-compile-time-annotation-for-jdk6-and-above).

> A maven plugin to process compile time annotation for jdk6 and above.

Can the `--module-path` support be conditional similar to [add support for -source and -target compiler options that are filled by maven.compiler.* properties]( https://github.com/bsorrentino/maven-annotation-plugin/commit/18212a3518d162ef6b21490960048090d2538132)?

**Comment by bsorrentino** [comment]
Hi @cordahi 

Thanks for your feedback.  The intention was to support from JDK9 and above but It is not stated in doc and probably it is not the right time, then I consider it a bug

Let me fix it

**Comment by bsorrentino** [comment]
I've released `4.0-jdk8` with required backward compatibility

**Comment by cordahi** [comment]
Sorry @bsorrentino, the new `4.0-jdk8` tag contains a problem.  It uses [compiler.isSupportedOption("--module-path")](https://github.com/bsorrentino/maven-annotation-plugin/blob/4.0-jdk8/processor/src/main/java/org/bsc/maven/plugin/processor/AbstractAnnotationProcessorMojo.java#L513) which throws an [UnsupportedOperationException](https://github.com/bsorrentino/maven-annotation-plugin/blob/4.0-jdk8/processor/src/main/java/org/bsc/maven/plugin/processor/AnnotationProcessorCompiler.java#L490).

```
[DEBUG] PROCESSOR COMPILER FORKED!
[ERROR] error on execute: use -X to have details 
[DEBUG] 
java.lang.UnsupportedOperationException: Not supported yet.
    at org.bsc.maven.plugin.processor.AnnotationProcessorCompiler.isSupportedOption (AnnotationProcessorCompiler.java:490)
    at org.bsc.maven.plugin.processor.AbstractAnnotationProcessorMojo.prepareOptions (AbstractAnnotationProcessorMojo.java:513)
    at org.bsc.maven.plugin.processor.AbstractAnnotationProcessorMojo.executeWithExceptionsHandled (AbstractAnnotationProcessorMojo.java:766)
    at org.bsc.maven.plugin.processor.AbstractAnnotationProcessorMojo.execute (AbstractAnnotationProcessorMojo.java:438)
```

**Comment by bsorrentino** [comment]
Hi @cordahi thanks for the feedback 

I'm going to verify the reason

**Comment by bsorrentino** [comment]
@cordahi I've deployed version `4.1.jdk8-SNAPSHOT` could you check if the fix works

thanks in advance

**Comment by cordahi** [comment]
Thanks @bsorrentino, `4.1.jdk8-SNAPSHOT` works great in my JDK8 project and addresses issue #81 to fail my build when there are errors

```
[INFO] --- maven-processor-plugin:4.1.jdk8-SNAPSHOT:process (process-dagger) @ apollo-helios ---

<trimmed plugin debug details and error messages>

4 errors
[ERROR] error on execute: use -X to have details 
[DEBUG] 
java.lang.Exception: error during compilation
    at org.bsc.maven.plugin.processor.AbstractAnnotationProcessorMojo.executeWithExceptionsHandled (AbstractAnnotationProcessorMojo.java:787)
    at org.bsc.maven.plugin.processor.AbstractAnnotationProcessorMojo.execute (AbstractAnnotationProcessorMojo.java:438)

<trimmed more stack traces>

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

**Comment by bsorrentino** [comment]
fix released in `4.1-jdk8` version

**Comment by tandr** [comment]
Will there be a 4.4-jdk8 or this has stopped with 4.2?

**Comment by bsorrentino** [comment]
Hi @tandr 

I've not released 4.4-jdk8 yet, because I plan it based on requests.

I'll work on it soon

**Comment by bsorrentino** [comment]
version `4.4-jdk8` released
